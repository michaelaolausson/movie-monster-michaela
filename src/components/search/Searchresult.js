import React from 'react';
import { Col, Row, Alert } from 'react-bootstrap'
import Movie from '../movie/Movie'


class Searchresult extends React.Component {

  render() {
    if (this.props.movies.results) {
      return(
        <Row>
        {this.props.movies.results.map((movie, index) => (
          <Movie id={movie.id} key={index} title={movie.title} year={movie.release_date} description={movie.overview} imagePath={movie.poster_path } />
      ))}
        </Row>
      )
    } else if (this.props.movies.results && this.props.movies.results.length === 0) {
       return(
        <Row>
          <Col lg={6}>
            <Alert variant="warning">
              Zero results. Zorry!
            </Alert>
          </Col>
        </Row>
      ) 
    } else {
      return (
        <Row></Row>
      )
    }
  }
}

export default Searchresult;
