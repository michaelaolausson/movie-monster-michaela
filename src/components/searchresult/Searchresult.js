import React from 'react';
import { ListGroup } from 'react-bootstrap'
import Movie from '../movie/Movie'


class Movielist extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {

    if (this.props.movies.results) {
      return(
        <ListGroup>
        {this.props.movies.results.map((movie, index) => (
          <Movie i={index} title={movie.title} year={movie.release_date} description={movie.overview} imagePath={movie.poster_path } />
      ))}
        </ListGroup>
      )

    } else {
      return (
        <ListGroup>
          <p className="text-muted">It's empty here!</p>
        </ListGroup>
      )
    }
  }
}

export default Movielist;
