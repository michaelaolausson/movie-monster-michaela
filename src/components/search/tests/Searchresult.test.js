import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow, mount } from 'enzyme';
import Searchresult from '../Searchresult.js'
import MOCK_SEARCH_SUCCESS from './data/mock_search-success'

describe('<Searchresult />', () => {
    const wrapper = mount(<Searchresult movies={MOCK_SEARCH_SUCCESS} />)
    it("should match the snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
    it('renders all 4 results', () => {
        expect(wrapper.find('.media').length).toBe(4)
    })
})