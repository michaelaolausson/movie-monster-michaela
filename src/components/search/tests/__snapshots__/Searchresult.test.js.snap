// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searchresult /> should match the snapshot 1`] = `"<div class=\\"row\\"><div class=\\"mb-4 col-xl-4 col-md-6 col-sm-12\\"><div class=\\"media\\"><img width=\\"100px\\" src=\\"https://image.tmdb.org/t/p/w200//cezWGskPY5x7GaglTTRN4Fugfb8.jpg\\" alt=\\"The Avengers - movie poster\\"><div class=\\"p-3 d-flex flex-wrap media-body\\"><div><h3>The Avengers</h3><p class=\\"text-muted p-0 mb-2\\"><small>Release date: 2012-04-25</small></p><p class=\\"p-0 mb-3\\">When an unexpected enemy emerges and threatens global safety and security, Nick Fury, director of the international peacekeeping agency known as S.H.I.E.L.D., finds himself in need of a team to pull the world back from the brink of disaster. Spanning the globe, a daring recruitment effort begins!...</p></div><div class=\\"align-self-end w-100\\"><button id=\\"add-to-favourites-btn\\" disabled=\\"\\" movieid=\\"24428\\" title=\\"Log in to add favourite\\" type=\\"button\\" class=\\"btn-block text-left btn btn-primary\\"><i class=\\"fa fa-heart-o\\"></i> Add to favourites</button><button id=\\"add-to-watchlist-btn\\" disabled=\\"\\" movieid=\\"24428\\" title=\\"Log in to add to watchlist\\" type=\\"button\\" class=\\"btn-block text-left btn btn-secondary\\"><i class=\\"fa fa-clock-o\\"></i> Add to watchlist</button></div></div></div></div><div class=\\"mb-4 col-xl-4 col-md-6 col-sm-12\\"><div class=\\"media\\"><img width=\\"100px\\" src=\\"https://image.tmdb.org/t/p/w200//or06FN3Dka5tukK1e9sl16pB3iy.jpg\\" alt=\\"Avengers: Endgame - movie poster\\"><div class=\\"p-3 d-flex flex-wrap media-body\\"><div><h3>Avengers: Endgame</h3><p class=\\"text-muted p-0 mb-2\\"><small>Release date: 2019-04-24</small></p><p class=\\"p-0 mb-3\\">After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what co...</p></div><div class=\\"align-self-end w-100\\"><button id=\\"add-to-favourites-btn\\" disabled=\\"\\" movieid=\\"299534\\" title=\\"Log in to add favourite\\" type=\\"button\\" class=\\"btn-block text-left btn btn-primary\\"><i class=\\"fa fa-heart-o\\"></i> Add to favourites</button><button id=\\"add-to-watchlist-btn\\" disabled=\\"\\" movieid=\\"299534\\" title=\\"Log in to add to watchlist\\" type=\\"button\\" class=\\"btn-block text-left btn btn-secondary\\"><i class=\\"fa fa-clock-o\\"></i> Add to watchlist</button></div></div></div></div><div class=\\"mb-4 col-xl-4 col-md-6 col-sm-12\\"><div class=\\"media\\"><img width=\\"100px\\" src=\\"https://image.tmdb.org/t/p/w200//7WsyChQLEftFiDOVTGkv3hFpyyt.jpg\\" alt=\\"Avengers: Infinity War - movie poster\\"><div class=\\"p-3 d-flex flex-wrap media-body\\"><div><h3>Avengers: Infinity War</h3><p class=\\"text-muted p-0 mb-2\\"><small>Release date: 2018-04-25</small></p><p class=\\"p-0 mb-3\\">As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use...</p></div><div class=\\"align-self-end w-100\\"><button id=\\"add-to-favourites-btn\\" disabled=\\"\\" movieid=\\"299536\\" title=\\"Log in to add favourite\\" type=\\"button\\" class=\\"btn-block text-left btn btn-primary\\"><i class=\\"fa fa-heart-o\\"></i> Add to favourites</button><button id=\\"add-to-watchlist-btn\\" disabled=\\"\\" movieid=\\"299536\\" title=\\"Log in to add to watchlist\\" type=\\"button\\" class=\\"btn-block text-left btn btn-secondary\\"><i class=\\"fa fa-clock-o\\"></i> Add to watchlist</button></div></div></div></div><div class=\\"mb-4 col-xl-4 col-md-6 col-sm-12\\"><div class=\\"media\\"><img width=\\"100px\\" src=\\"https://image.tmdb.org/t/p/w200//t90Y3G8UGQp0f0DrP60wRu9gfrH.jpg\\" alt=\\"Avengers: Age of Ultron - movie poster\\"><div class=\\"p-3 d-flex flex-wrap media-body\\"><div><h3>Avengers: Age of Ultron</h3><p class=\\"text-muted p-0 mb-2\\"><small>Release date: 2015-04-22</small></p><p class=\\"p-0 mb-3\\">When Tony Stark tries to jumpstart a dormant peacekeeping program, things go awry and Earthâ€™s Mightiest Heroes are put to the ultimate test as the fate of the planet hangs in the balance. As the villainous Ultron emerges, it is up to The Avengers to stop him from enacting his terrible plans, and soo...</p></div><div class=\\"align-self-end w-100\\"><button id=\\"add-to-favourites-btn\\" disabled=\\"\\" movieid=\\"99861\\" title=\\"Log in to add favourite\\" type=\\"button\\" class=\\"btn-block text-left btn btn-primary\\"><i class=\\"fa fa-heart-o\\"></i> Add to favourites</button><button id=\\"add-to-watchlist-btn\\" disabled=\\"\\" movieid=\\"99861\\" title=\\"Log in to add to watchlist\\" type=\\"button\\" class=\\"btn-block text-left btn btn-secondary\\"><i class=\\"fa fa-clock-o\\"></i> Add to watchlist</button></div></div></div></div></div>"`;
